src/Simulation/Simulation.jl

```diff
@@ -928,7 +901,7 @@ function _calculate_potential!( sim::Simulation{T, CS}, potential_type::UnionAll
                                                 n_iterations_between_checks = n_iterations_between_checks,
                                                 max_n_iterations = max_n_iterations,
                                                 depletion_handling = depletion_handling,
-                                                use_nthreads = use_nthreads,
+                                                use_nthreads = guess_nt ? _guess_optimal_number_of_threads_for_SOR(size(sim.electric_potential.grid), max_nthreads[iref+1], CS) : max_nthreads[iref+1],
```

Comment:
Will it increase the performance to call  `_guess_optimal_number_of_threads_for_SOR` once at the beginning for all `iref` using broadcast?

Reply:
Correct, we need the size of the grid.